Sources/Moya/MoyaError.swift

```diff
@@ -16,7 +16,7 @@ public enum MoyaError: Swift.Error {
     case objectMapping(Swift.Error, Response)
 
     /// Indicates that Encodable couldn't be encoded into Data
-    case encodableMapping(Swift.Error)
+    case encodableMapping(String)
```

Comment:
Why does this changed to a `String`?

Reply:
I changed it for a String because now it's only thrown when using a CustomParams with a wrong encoder.
But as you pointed it, i've read again the way parameters are encoded and it should definitively also still be thrown with the new way of encoding parameters (Endpoint.swift:L87). If I do so the associated value would be an Error again.
I would also need a new `MoyaError` with a `String` for missuses of `CustomParams`. Any suggestion for its name?

Reply:
Another option would be to have `MoyaError.encodableMapping(Swift.Error?, String?)`

Reply:
> Another option would be to have `MoyaError.encodableMapping(Swift.Error?, String?)`

I personally don't like this approach because it is not clear what this string means for the user. I mean it is probably better to have something like a custom moya error or even better, extend the MoyaError to have a localized description property e.g. NSError ... so that kind of message could go there :) What you think?

Reply:
I've pushed a new dedicated error so that `encodableMapping` is left untouched.